engine/src/main/java/com/odysseusinc/arachne/executionengine/execution/r/DockerOverseer.java

```diff
@@ -54,21 +57,48 @@ public DockerOverseer(
                 return (exitCode == 0)
                         ? new ExecutionOutcome(Stage.COMPLETED, null, out)
                         : new ExecutionOutcome(Stage.EXECUTE, "Exit code " + exitCode, out);
-            }
+            } 
         }));
         pos = stdout.length();
         this.client = client;
-        init.thenAccept(containerId ->
-                executor.scheduleWithFixedDelay(this::writeLogs, updateInterval, updateInterval, TimeUnit.MILLISECONDS)
-        );
+        init.thenAccept(containerId -> {
+            executor.scheduleWithFixedDelay(this::writeLogs, updateInterval, updateInterval, TimeUnit.MILLISECONDS);
+        });
         this.init = init;
     }
 
     @Override
     public CompletableFuture<ExecutionOutcome> abort() {
-        return result.isDone() ? result : CompletableFuture.completedFuture(
-                new ExecutionOutcome(Stage.ABORT, "Abort is not yet supported for docker analysis", null)
-        );
+        if (!init.isDone()) {
+            init.cancel(true);
+        }
+
+        return init.handle((containerId, throwable) -> {
+            if (containerId == null) {
+                if (throwable != null) {
+                    log.error("Error during initialization: {}", throwable.getMessage());
+                    return new ExecutionOutcome(Stage.ABORTED, "Initialization failed: " + throwable.getMessage() + "\n" + stdout, "Initialization failed: " + throwable.getMessage() + "\n" + stdout);
+                } else {
+                    return new ExecutionOutcome(Stage.ABORTED, null, "Container initialization was canceled:\n" + stdout);
+                }
+            } else {
+                if (result.isDone()) {
+                    return result.join();
+                } else {
+                    try {
+                        client.stopContainerCmd(containerId).exec();
+                        return new ExecutionOutcome(Stage.ABORTED, null, "Docker container aborted successfully:\n" + stdout);
+                    } catch (NotFoundException e) {
+                        log.info("Container not found or already stopped: " + e.getMessage());
+                        return new ExecutionOutcome(Stage.COMPLETED, "Container not found or already stopped: " + e.getMessage(), "Container not found or already stopped: " + e.getMessage() + "\n" + stdout);
```

Comment:
Always reporting COMPLETED + error is incorrect here. 
Could have completed successfully or failed. Use result.join() instead.